### DISCUSSION TITLE
Error when pushing in Github Desktop

### DISCUSSION THREAD

**Discussion by iriseel** [community/community#23396] [New to GitHub]
<p>Hi all, I’ve been using Github + Github Desktop at a super basic level for the past few months — basically only creating repos, committing to main, and pushing updates. That’s been sufficient for my purposes, but over a week ago, after “Committing to main” I tried to “Push origin” on a repo, and instead received an error message, saying:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/2/b2ccfaed9f3064f6cf76ecb083d44c0a3e552f38.png" data-download-href="/uploads/short-url/pvK9MswOp8qrV0msVLEmoVK06P6.png?dl=1" title="Github Desktop Error Message" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/74837161/181076735-bef0fe11-fcbe-4ed9-997b-9106451ba9ee.png" alt="Github Desktop Error Message" data-base62-sha1="pvK9MswOp8qrV0msVLEmoVK06P6" width="690" height="413" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/2/b2ccfaed9f3064f6cf76ecb083d44c0a3e552f38_2_690x413.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/2/b2ccfaed9f3064f6cf76ecb083d44c0a3e552f38_2_1035x619.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/b/2/b2ccfaed9f3064f6cf76ecb083d44c0a3e552f38_2_1380x826.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/2/b2ccfaed9f3064f6cf76ecb083d44c0a3e552f38_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Github Desktop Error Message</span><span class="informations">2106×1262 249 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
(Note: Every so often, the number that appears after “ERROR: Couldn’t connect to 127.0.01:” changes.)
<p>Since then, I turned off my firewall and flailed down various online rabbit holes, but to no avail. I made a new repo just to test things out and I can’t “publish branch” either because I get the same error message when I try. I hadn’t done anything different from usual leading up to that push that brought up the first fateful error message. I have no idea what most of this error message even means; I’m at a total loss as to what went wrong here. Does anyone have any idea what could be going on?</p>

**Comment by bhawesh2002** [comment, marked as answer]
<p><a class="mention" href="/iriseel">@iriseel</a><br>
I might be wrong with this but I never faced the same issue while using github desktop.<br>
The last line of the error mesage is<br>
<strong><code>fatal: coult not read Username for 'https://github.com': terminal prompts disabled</code></strong></p>
<p>the line <strong><code>terminal prompts disabled</code></strong> is the main issue.<br>
Tried searching for the issue on googlle and this is what I found</p>
<p><strong>1.</strong></p><aside class="onebox stackexchange" data-onebox-src="https://stackoverflow.com/questions/32232655/go-get-results-in-terminal-prompts-disabled-error-for-github-private-repo">
  <header class="source">

      <a href="https://stackoverflow.com/questions/32232655/go-get-results-in-terminal-prompts-disabled-error-for-github-private-repo" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>

  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/478311/tomcam" target="_blank" rel="noopener nofollow ugc">
    <img alt="tomcam" src="https://user-images.githubusercontent.com/55053472/181076737-1b178a34-f5bb-49cd-951c-6839af45d4e0.jpeg" class="thumbnail onebox-avatar" width="128" height="128">
  </a>

<h4>
  <a href="https://stackoverflow.com/questions/32232655/go-get-results-in-terminal-prompts-disabled-error-for-github-private-repo" target="_blank" rel="noopener nofollow ugc">go get results in 'terminal prompts disabled' error for github private repo</a>
</h4>

<div class="tags">
  <strong>git, go</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/478311/tomcam" target="_blank" rel="noopener nofollow ugc">
    tomcam
  </a>
  on <a href="https://stackoverflow.com/questions/32232655/go-get-results-in-terminal-prompts-disabled-error-for-github-private-repo" target="_blank" rel="noopener nofollow ugc">05:10PM - 26 Aug 15 UTC</a>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p><strong>2.</strong></p><aside class="onebox allowlistedgeneric" data-onebox-src="https://jacopretorius.net/2018/05/git-error-could-not-read-username.html">
  <header class="source">

      <a href="https://jacopretorius.net/2018/05/git-error-could-not-read-username.html" target="_blank" rel="noopener nofollow ugc">Jaco Pretorius</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/360;"><img src="https://user-images.githubusercontent.com/55053472/181076738-5fcd300a-4bc5-4538-8228-6141e42662bf.jpeg" class="thumbnail" width="690" height="360" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cfc7341ae322e14221886dc93d8eb163d544fbc7_2_690x360.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cfc7341ae322e14221886dc93d8eb163d544fbc7_2_1035x540.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/f/cfc7341ae322e14221886dc93d8eb163d544fbc7.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cfc7341ae322e14221886dc93d8eb163d544fbc7_2_10x10.png"></div>

<h3><a href="https://jacopretorius.net/2018/05/git-error-could-not-read-username.html" target="_blank" rel="noopener nofollow ugc">Git Error: Could could not read Username for 'https://github.com': terminal...</a></h3>

  <p>Today I ran into this error while trying to update Homebrew.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I think the 2nd solution is more eazy.<br>
Hope it Helps :rescue_worker_helmet:</p>

**Comment by iriseel** [comment]
<p><a class="mention" href="/bhawesh2002">@bhawesh2002</a> Thanks for your reply! I took a look at what you found and tried the second option of adding the following code to terminal:</p>
<pre><code class="lang-auto">git config --global --add url."git@github.com:".insteadOf "https://github.com/"
</code></pre>
<p>I also created an SSH key as per the instructions <a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent" rel="noopener nofollow ugc">here</a> and linked it to my github account. However, I hadn’t yet tested my connection as instructed <a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/testing-your-ssh-connection" rel="noopener nofollow ugc">here</a>. So what I got when I tried to push in Github Desktop was the more usual Authentication error message, “Authentication failed. Some common reasons include …” instead of the original one. Afterwards, I doubled back and tested my SSH connection, after which the push on Github Desktop went through!!! I could also “publish branch” the new repo I made. Hurray!</p>
<p>My only nagging worry now is that I still have no idea why I ran into the original error in the first place. Unlike what the author of the second troubleshooting option you suggested wrote, I didn’t have 2FA enabled. And unlike the op of the first Stackoverflow option, I wasn’t doing anything in Terminal before this problem — I had only been pushing and pulling in Github Desktop.</p>

**Comment by bhawesh2002** [comment]
<p><a class="mention" href="/iriseel">@iriseel</a><br>
Happy that your error is solved.</p>
<p>And I also don’t have any idea on why you encountered the original error.<br>
I think it is one of those random errors that we get when everything goes fine. :wink:</p>
